src/reports/components/fast-pass-report-summary.tsx

```diff
@@ -1,24 +1,86 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
+import { requirements } from 'DetailsView/components/tab-stops/requirements';
+import { TabStopsFailedCounter } from 'DetailsView/tab-stops-failed-counter';
 import * as React from 'react';
 import { FastPassReportResultData } from 'reports/components/fast-pass-report';
 import { OutcomeSummaryBar } from 'reports/components/outcome-summary-bar';
 import { OutcomeStats } from 'reports/components/outcome-type';
 import { RequirementOutcomeType } from 'reports/components/requirement-outcome-type';
 
+export type FastPassReportSummaryDeps = {
+    tabStopsFailedCounter: TabStopsFailedCounter;
+};
+
 export interface FastPassReportSummaryProps {
+    deps: FastPassReportSummaryDeps;
     results: FastPassReportResultData;
 }
 
 export const allOutcomeTypes: RequirementOutcomeType[] = ['fail', 'incomplete', 'pass'];
 
 export class FastPassReportSummary extends React.Component<FastPassReportSummaryProps> {
     public render(): JSX.Element {
+        const tabResults = [];
+
+        for (const [requirementId, data] of Object.entries(this.props.results.tabStops)) {
+            if (data.status !== 'fail') {
+                continue;
+            }
+
```

Comment:
I think it might be more straightforward to just do this loop once. Inside the loop you can check the `data.status` and push to the correct array of results, rather than looping 3 times and checking for the correct status in each one.